packages/agreement-process/src/services/agreementService.ts

```diff
@@ -427,6 +430,28 @@ export function agreementServiceBuilder(
 
       return agreement;
     },
+    async computeAgreementsStateByAttribute(
+      attributeId: AttributeId,
+      consumer: CompactTenant,
+      { logger, correlationId }: WithLogger<AppContext>
+    ): Promise<void> {
+      logger.info(
+        `Recalculating agreements state for attribute ${attributeId}`
+      );
+
+      const events = await computeAgreementsStateByAttribute(
+        attributeId,
+        consumer,
+        agreementQuery,
+        eserviceQuery,
+        correlationId,
+        logger
+      );
+
+      for (const event of events) {
+        await repository.createEvent(event);
+      }
```

Comment:
All events are for different agreements i.e., different stream ids, no need for a transaction here